### DISCUSSION TITLE
Error "IOException: The response ended prematurely" when sending an https request.

### DISCUSSION THREAD

**Discussion by tushinivan** [dotnet/yarp#1388] [Q&A] [answered]
Hello! I'm trying to make an https request to https://example.com, but I get a 

System.Net.Http.HttpRequestException: "An error occurred while sending the request." 
IOException: The response ended prematurely.

```
HttpClient cl = new HttpClient(new HttpClientHandler() 
            { 
                AllowAutoRedirect = false, 
                Proxy = new WebProxy("localhost:5001") 
            });
var r = await cl.GetAsync("https://example.com/");

```
as an instance of YARP I use the example [ReverseProxy.Direct.Sample](https://github.com/microsoft/reverse-proxy/tree/main/samples/ReverseProxy.Direct.Sample), can you please tell me if additional YARP configuration is required?

**Comment by Tratcher** [comment, marked as answer]
YARP is not a forward proxy that can be used in the WebProxy field. This may help explain the difference: https://www.linuxbabe.com/it-knowledge/differences-between-forward-proxy-and-reverse-proxy

You'd send the request to YARP like this:
```
HttpClient cl = new HttpClient();
var r = await cl.GetAsync("https://localhost:5001");
```
And YARP would use its configuration decide if that request should be sent to example.com or somewhere else.
